docs/support/tutorials/python-usage-guide.md

```diff
@@ -0,0 +1,254 @@
+	  	including how to convert Docker containers to Apptainer containers.
+
+	The [CSC conda tutorial](./conda.md) describes in more detail
+	what conda is and how to use it. Some parts of the tutorial may
+	also be helpful for Tykky installations.
+
+## Python development environments
+
+Python code can be edited with a console-based [text editor directly on the
+supercomputer](./env-guide/text-and-image-processing.md).
+Codes can also be edited on your local machine and copied to the supercomputer
+with [scp](../../data/moving/scp.md) or [graphical file transfer
+tools](../../data/moving/graphical_transfer.md).
+You can also [edit Python scripts in Puhti from your local
+PC](./remote-dev.md) with some code editors like Visual
+Studio Code.
+
+Finally, several graphical programming environments can be used directly on the
+supercomputer, such as Jupyter Notebooks, Spyder and Visual Studio Code, through
+the [Puhti web interface](../../computing/webinterface/index.md).
+
+### Jupyter Notebooks
+
+[Jupyter Notebooks](https://jupyter.org/) allows one to run Python code via a web
+browser running on a local PC. The notebooks can combine code, equations,
+visualizations and narrative text in a single document. Many of our modules, including
+[python-data](../../apps/python-data.md), the deep learning modules and
+[geoconda](../../apps/geoconda.md)
+include the Jupyter notebook package. See the tutorial [how to set up and connect to
+a Jupyter Notebook](./rstudio-or-jupyter-notebooks.md) for using
+Jupyter in CSC environment.
+
+### Spyder
+
+[Spyder](https://www.spyder-ide.org/) is scientific Python development
+environment. Modules [python-data](../../apps/python-data.md) and
+[geoconda](../../apps/geoconda.md)
+have Spyder included. The best option for using it is through the [Puhti
+web interface remote desktop](../../computing/webinterface/desktop.md).
+
+## Python parallel jobs
+
+Python has several different packages for parallel processing:
+
+* [multiprocessing](https://docs.python.org/3/library/multiprocessing.html)
+* [joblib](https://joblib.readthedocs.io/en/latest/)
+* [dask](https://docs.dask.org)
+* [mpi4py](https://mpi4py.readthedocs.io) - Python interface to MPI
+
+The `multiprocessing` package is likely the easiest to use and as it is part of the
+Python standard library it is included in all Python installations. `joblib` provides
+some more flexibility. `multiprocessing` and `joblib` are suitable for one
+node (max 40 cores). `dask` is the most versatile and has several options for
+parallelization. Please see [CSC's Dask tutorial](./dask-python.md)
+which includes both single-node (max 40 cores) and multi-node examples.
+
+See our GitHub repository for some [examples for using the different parallelization
+options with Puhti](https://github.com/csc-training/geocomputing/tree/master/python/puhti).
+
+The `mpi4py` is not included in the current Python environments in CSC supercomputers,
```

Comment:
Actually pytorch has mpi4py. Maybe it should be added to python-data in the future?

Reply:
The section now mentions that `mpi4py` is included in the `pytorch` module.